openmetadata-service/src/main/java/org/openmetadata/service/search/SearchListFilter.java

```diff
@@ -287,4 +284,17 @@ private String getTestSuiteIdCondition(String testSuiteId) {
         "{\"nested\":{\"path\":\"testSuites\",\"query\":{\"term\":{\"testSuites.id\":\"%s\"}}}}",
         testSuiteId);
   }
+
+  private String getTestCaseTypeCondition(String type, String field) {
+    return switch (type) {
+      case Entity.TABLE -> String.format(
+          "{\"bool\": {\"must_not\": [{\"regexp\": {\"%s\": \".*::columns::.*\"}}]}}", field);
```

Comment:
@TeddyCr we can use QueryBuilders here, is it because of the package issue between es/os we are using string queries?